youtube_dl/extractor/camwhoresbay.py

```diff
@@ -0,0 +1,141 @@
+            # 'formats': [{
+            #     'url': r're:https://www\.camwhoresbay\.com/get_file/7/55259a27805bf1313318c14b2afb0dae1fef6e1dd4/484000/484472/484472\.mp4/\?rnd=.+',
+            #     'height': 360
+            # }, {
+            #     'url': r're:https://www\.camwhoresbay\.com/get_file/7/55259a27805bf1313318c14b2afb0dae1fef6e1dd4/484000/484472/484472_480p\.mp4/\?rnd=.+',
+            #     'height': 480
+            # }, {
+            #     'url': r're:https://www\.camwhoresbay\.com/get_file/7/55259a27805bf1313318c14b2afb0dae1fef6e1dd4/484000/484472/484472_720p\.mp4/\?rnd=.+',
+            #     'height': 720
+            # }]
+            'thumbnails': [{
+                'url': 'https://cwbstatic.cdntrex.com/contents/videos_screenshots/484000/484472/preview.mp4.jpg',
+                'height': 360
+            }, {
+                'url': 'https://cwbstatic.cdntrex.com/contents/videos_screenshots/484000/484472/preview_480p.mp4.jpg',
+                'height': 480
+            }, {
+                'url': 'https://cwbstatic.cdntrex.com/contents/videos_screenshots/484000/484472/preview_720p.mp4.jpg',
+                'height': 720
+            }],
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
+        }
+    }
+
+    @staticmethod
+    def _parse_formats(flashvars):
+        formats = []
+        rnd = flashvars.get('rnd')
+        for k, v in flashvars.items():
+            if re.match(r"^video_(alt_)?url\d?$", k):
+                f = {
+                    'url': '{}?rnd={}'.format(v, rnd),
+                    'height': int(flashvars['{}_text'.format(k)][:-1])
+                }
+                formats.append(f)
+        return formats
+
+    @staticmethod
+    def _parse_thumbnails(flashvars):
+        thumbnails = []
+        for k, v in flashvars.items():
+            if re.match(r"^preview_url\d$", k):
+                t = {
+                    'url': 'https:{}'.format(v),
+                    'height': int(flashvars[k.replace('_url', '_height')])
+                }
+                thumbnails.append(t)
+        return thumbnails
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_regex(r'<div class=\"headline\">\n\s*<h1>(.+?)</h1>', webpage, 'title')
+
+        flashvars = self._parse_json(
+            self._search_regex(r'var\s+flashvars\s+=\s+({\n.+});', webpage, 'flashvars', default='{}'),
+            video_id, transform_source=js_to_json)
+        uploader = self._search_regex(r'class=\"avatar\" href=\"http.+/\" title=\"(.+)\"', webpage, 'uploader',
```

Comment:
No need to escape " inside r'...'. As before, relax the pattern:
```suggestion
        uploader = self._search_regex(r'''(?s)\bclass\s*=\s*["']avatar["'][^>]+?\bhref\s*=\s*["']http.+/["'][^>]+?\btitle\s*=\s*["'](.+)["']''', webpage, 'uploader',
```
Omit matching the `href` attribute, if the page has only one tag with class `avatar`, or the first one is wanted:
```suggestion
        uploader = self._search_regex(r'''(?s)\bclass\s*=\s*["']avatar["'][^>]+?\btitle\s*=\s*["'](.+)["']''', webpage, 'uploader',
```
Or match the entire tag that contains the attributes and use `extract_attributes(tag).get('title')`